### PR TITLE
Allow scope of Remove Empty Lines to be limited by an active selection

### PR DESCRIPTION
**Pull request by sasumner**
Implements #8428

### CODE REVIEW

File: PowerEditor/src/Notepad_plus.cpp [hunk 1]
```diff
@@ -1400,10 +1400,13 @@ void Notepad_plus::removeEmptyLine(bool isBlankContained)
 		env._str2Search = TEXT("^$(\\r\\n|\\r|\\n)");
 	}
 	env._str4Replace = TEXT("");
-    env._searchType = FindRegex;
-
-	_findReplaceDlg.processAll(ProcessReplaceAll, &env, true);
-
+	env._searchType = FindRegex;
+	auto mainSelStart = _pEditView->execute(SCI_GETSELECTIONSTART);
```
Comment (sasumner) [inline PowerEditor/src/Notepad_plus.cpp:1404]: Is it nicer to do this here instead?
`Sci_CharacterRange mainSelection = _pEditView->getSelection();`
And then work with that in the next lines?
Is it dangerous to comment on one's own code?  Is that asking for trouble? :-)

Comment (donho) [reply PowerEditor/src/Notepad_plus.cpp:1404]: > Is it nicer to do this here instead?:
Sci_CharacterRange mainSelection = _pEditView->getSelection();

Yes, why not?

> Is it dangerous to comment on one's own code? Is that asking for trouble? :-)

Is it a philosophical question? Am I supposed to answer this question?  :)

**Comment by donho** [comment]
PR tested and it works as usual.
However, there's a behaviour you might not consider:
If a word or some words of the same line is/are selected, do we treat the selected zone in one line or we ignore this case (so treat the whole document) for line*s* operation?

**Comment by donho** [comment]
@sasumner Any thought about the behaviour mentioned above?

**Comment by donho** [comment]
@sasumner Any thought about the behaviour mentioned above?

**Comment by sasumner** [comment]
@donho 
I'm typing my response.
Give me a time to complete it please. :-)

**Comment by sasumner** [comment]
@donho

>If a word or some words of the same line is/are selected, 

Overall, it might be strange for users until they learn about selection-based removal of empty lines...

>do we treat the selected zone in one line 

Nothing would change for the user in this case; it would look like their command didn't do anything.
That's OK because they would think "What did I do wrong?" and then "Oh, this must now be a selection-based command".

>or we ignore this case (so treat the whole document) for lines operation?

I would think NO, we do not treat the whole document here.
User must give up the selection completely in order to treat the whole document.
Or, of course, do ctrl+a first.

PR is OK as it is for this functionality.

But if you want changes, just tell me what they are.

**Comment by donho** [comment]
User could double click a word accidentally (that makes word selection), then scroll down, and run this command. In this case, no empty line is removed, and user is confused. That's the scenario that I can think about. 

> But if you want changes, just tell me what they are.

Just want to get your thought about it.

**Comment by sasumner** [comment]
>User could ... makes ... selection, then scroll down, and run this command. In this case, no empty line is removed, and user is confused.

I agree that is a possible scenario.
If adopted, the new behavior could be strange to users for a bit of time, until they are exposed to it and become familiar.

**Comment by donho** [comment]
OK, let's do it in this way currently. We will see if there are some users complain it in the future.

**Pull request closed without merging**